### PR TITLE
reverse-proxy: T6409: Remove unused backend parameters

### PR DESCRIPTION
**Pull request by Embezzle**
## Change Summary

There is an unused backend `parameters http-check` configuration node in reverse-proxy backends.
HTTP health-checks are available via separate `backend http-check` node implemented in [T6246](https://vyos.dev/T6246).

We should remove redundant node, it has no effect on rendered `haproxy.cfg` configuration file so is pointless.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [x] Other (please describe): Feature/functionality removal

## Related Task(s)

[https://vyos.dev/T6409](https://vyos.dev/T6409)

## Related PR(s)

* https://github.com/vyos/vyos-documentation/pull/1469

## Component(s) name

- load-balance -> reverse-proxy

## Smoketest result

```
vyos@vyos:~$ python3 /usr/libexec/vyos/tests/smoke/cli/test_load-balancing_reverse-proxy.py
test_01_lb_reverse_proxy_domain (__main__.TestLoadBalancingReverseProxy.test_01_lb_reverse_proxy_domain) ... ok
test_02_lb_reverse_proxy_cert_not_exists (__main__.TestLoadBalancingReverseProxy.test_02_lb_reverse_proxy_cert_not_exists) ...
PKI does not contain any certificates!


Certificate "cert" not found in configuration!

ok
test_03_lb_reverse_proxy_ca_not_exists (__main__.TestLoadBalancingReverseProxy.test_03_lb_reverse_proxy_ca_not_exists) ...
PKI does not contain any CA certificates!


CA Certificate "ca-test" not found in configuration!

ok
test_04_lb_reverse_proxy_backend_ssl_no_verify (__main__.TestLoadBalancingReverseProxy.test_04_lb_reverse_proxy_backend_ssl_no_verify) ...
backend bk-01 cannot have both ssl options no-verify and ca-certificate
set!

ok
test_05_lb_reverse_proxy_backend_http_check (__main__.TestLoadBalancingReverseProxy.test_05_lb_reverse_proxy_backend_http_check) ... ok
test_06_lb_reverse_proxy_tcp_mode (__main__.TestLoadBalancingReverseProxy.test_06_lb_reverse_proxy_tcp_mode) ... ok
test_07_lb_reverse_proxy_http_response_headers (__main__.TestLoadBalancingReverseProxy.test_07_lb_reverse_proxy_http_response_headers) ...
service https_front must be set to http mode to use
http_response_headers!

ok

----------------------------------------------------------------------
Ran 7 tests in 30.645s

OK
```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly

### CODE REVIEW

**Comment by sever-sever** [comment]
Is it better to fix the template to get it working?
You cannot delete nodes without migration scripts

**Comment by Embezzle** [comment]
> You cannot delete nodes without migration scripts

Apologies, I have added a migration script.

> Is it better to fix the template to get it working?

Don't think that would make sense - identical functionality is already available in another configuration node.

I am open to maintainers suggestions on whether this migration script should just delete the removed node, or also migrate configurations to the working `http-check` node.
Thinking it best to just delete - this node has historically had no functionality, so migrating to the other node would introduce a change in behaviour...

**Comment by nvollmar** [comment]
`parameters http-check` doesn't seem to be evaluated anywhere/having an effect on the configuration.

The best way to handle it is probably just removing the old option to not break behavior.
Then make a hint in the release notes to users to check out the new option if they intended to use http checks.

btw, documenation also has to be updated.

File: src/migration-scripts/reverse-proxy/0-to-1 [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+with open(file_name, 'r') as f:
+    config_file = f.read()
+
+config = ConfigTree(config_file)
+base = ['load-balancing', 'reverse-proxy', 'backend']
+if not config.exists(base):
+    # Nothing to do
+    exit(0)
+else:
```
Comment (c-po) [inline src/migration-scripts/reverse-proxy/0-to-1]: No need for an `else:` path here as the conditional if has an exit.

Removing else would save one indention level.

**Review by c-po — CHANGES_REQUESTED** [review]

**Comment by dmbaturin** [comment]
@mergifyio backport sagitta

**Review by sever-sever — APPROVED** [review]

**Review by dmbaturin — APPROVED** [review]

**Review by c-po — APPROVED** [review]

**Pull request merged**